.github/workflows/beam_PreCommit_Java_PVR_Flink_Batch.yml

```diff
@@ -99,10 +99,9 @@ jobs:
         with:
           name: JUnit Test Results
           path: "**/build/reports/tests/"
-      - name: Publish JUnit Test Results
-        uses: EnricoMi/publish-unit-test-result-action@v2
-        if: always()
+      - name: Upload test report
+        uses: actions/upload-artifact@v3
         with:
-          commit: '${{ env.prsha || env.GITHUB_SHA }}'
-          comment_mode: ${{ github.event_name == 'issue_comment'  && 'always' || 'off' }}
-          files: '**/build/test-results/**/*.xml'
\ No newline at end of file
+          name: java-code-coverage-report
+          path: "**/build/test-results/**/*.xml"
+# TODO: Investigate 'Max retries exceeded' issue with EnricoMi/publish-unit-test-result-action@v2.
```

Comment:
this happens for jobs run on highmem cluster -- likely some firewall rule

https://github.com/apache/beam/blob/48ad7e338b7bc3aad8b1f873ba8b90c3f527b7be/.github/workflows/beam_PreCommit_Java_PVR_Flink_Batch.yml#L75